Content.Server/NukeOps/WarDeclaratorComponent.cs

```diff
@@ -13,7 +13,7 @@ public sealed partial class WarDeclaratorComponent : Component
     /// </summary>
     [ViewVariables(VVAccess.ReadWrite)]
     [DataField("message")]
-    public string Message;
+    public string? Message;
```

Comment:
Why was this done?

Reply:
Warning/error for Non nullable field that is not initialized, when i checked its usage it had a default message set from fluent loc variable in WarDeclaratorSystem.cs 
`        if (component.AllowEditingMessage && message != string.Empty)
        {
            component.Message = message;
        }
        else
        {
            message = Loc.GetString("war-declarator-default-message");
        }`
        
So I made it nullable as I didn't want to put a non localized string in to it as the default. I guess i could set the IDE to ignore it as an alternative if this isn't convention.

Reply:
Just to clarify, this is unrelated to this bug fix, correct? (It is helpful to separate other unrelated changes from the bug that you're fixing.)

Reply:
This is also technically a public API field type change that has the potential to break downstream code. I believe the impact here is small, but it's technically a breaking change. It is not good to have breaking changes in bug fixes.

Reply:
> Just to clarify, this is unrelated to this bug fix, correct? (It is helpful to separate other unrelated changes from the bug that you're fixing.)

Yeah its unrelated to the fix.

> This is also technically a public API field type change that has the potential to break downstream code. I believe the impact here is small, but it's technically a breaking change. It is not good to have breaking changes in bug fixes.

Was just in that area of the code base and saw it, I've removed it for now. A side question how do you know its used downstream/ API field or not?

Reply:
> Was just in that area of the code base and saw it, I've removed it for now. A side question how do you know its used downstream/ API field or not?

By checking the code for all forks that you know about and all the forks that you also don't know about :)

In short, since it's typically not possible to check, all breaking changes must be prominently documented. Since breaking changes can and do break compilation for forks, it is poor form to bundle breaking changes with bug fixes.

This is why object-oriented programming languages have visibility qualifiers (e.g. `public` and `private`). Public methods are by definition visible to other classes (are "public") and form part of the API. Do not change the API willy-nilly.